.github/workflows/rtl-repo-sync.yml

```diff
@@ -0,0 +1,87 @@
+name: RTL Repo Sync
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  update-rtl:
+    name: Update RTL Submodule
+    runs-on: ubuntu-22.04
+    outputs:
+      branch_name: ${{ steps.branch.outputs.branch }}
+      ci_checks_needed: ${{ steps.status.outputs.status }}
+    steps:
+      - name: Get current date
+        id: date
+        run: echo "date=$(date +'%Y%m%d')" >> $GITHUB_OUTPUT
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          ref: 'main'
+
+      - name: Update the Caliptra RTL submodule and rebuild registers
+        run: |
+          git submodule update --remote hw-latest/caliptra-rtl
+          cargo run --release --manifest-path registers/bin/generator/Cargo.toml -- hw-latest/caliptra-rtl registers/src/
+
+      - name: Find available branch name
+        id: branch
+        run: |
+          BRANCH_BASE=ci_rtl_${{ steps.date.outputs.date }}
+          INDEX=0
+          while git ls-remote --exit-code --heads origin ${BRANCH_BASE}_${INDEX}
+          do
+              ((INDEX+=1))
+          done
+          echo "branch=${BRANCH_BASE}_${INDEX}" >> $GITHUB_OUTPUT
+
+      - name: Run git status
+        id: status
+        run: echo "status=$(git status -s | head -1)" >> $GITHUB_OUTPUT
+
+      # If there's a new HW release (submodule has been updated), add a commit to main branch
+      - name: Commit changes
+        if: ${{ steps.status.outputs.status }}
+        run: |
+          git config --global user.name "GitHub CI"
+          git config --global user.email "username@users.noreply.github.com"
+          git checkout -b ${{ steps.branch.outputs.branch }}
+          git commit -a -m "CI: Update Caliptra RTL"
+          git push -u origin ${{ steps.branch.outputs.branch }}
+
+  build-test:
+    uses: ./.github/workflows/build-test.yml
+    needs: update-rtl
+    if: needs.update-rtl.outputs.ci_checks_needed
+    with:
+      ref: ${{ needs.update-rtl.outputs.branch_name }}
+
+  build-test-verilator:
+    uses: ./.github/workflows/build-test-verilator.yml
+    needs: update-rtl
+    if: needs.update-rtl.outputs.ci_checks_needed
+    with:
+      ref: ${{ needs.update-rtl.outputs.branch_name }}
+
+  merge-rtl-update:
+    name: Merge RTL Update
+    needs: [update-rtl, build-test, build-test-verilator]
+    if: needs.update-rtl.outputs.ci_checks_needed
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          ref: 'main'
+          fetch-depth: 0
+
+      - name: Merge RTL changes to main branch
+        run: |
+          git config --global user.name "GitHub CI"
+          git config --global user.email "username@users.noreply.github.com"
+          git merge origin/${{ needs.update-rtl.outputs.branch_name }}
```

Comment:
May fail here if there's a commit that comes in during the CI run before this

Reply:
Consider consolidating CI checks into a single workflow, so when somebody adds a new one, they add it to the single CI workflow, and then we can call it here. Make sure all CI checks show up independently in the PR.

Reply:
Also check if we can just create/manage/close a PR

Reply:
> May fail here if there's a commit that comes in during the CI run before this

Fixed this